### PR TITLE
Documentation Updates

### LINKED ISSUE #833: OBaaS Setup Images
ref orahub `#516` `#396` `#436` `#319`

### PR DESCRIPTION
**Pull request by gotsysdba**
Closes #833

### CODE REVIEW

File: docs-source/spring/content/infrastructure/database/_index.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+## Configuration
+
+1. During the configuration of the Oracle Backend for Spring Boot and Microservices, ensure that the **Edition** is set to **Standard**:
+
+    ![Standard Edition](../images/standard_edition.png "Standard Edition")
+
+1. Enable and Configure *Bring Your Own Virtual Network*
+
+1. Tick the Bring Your Own Database" checkbox and, depending on the *Bring Your Own Database - Type*, provide the appropriate values.
```
Comment (markxnelson) [inline docs-source/spring/content/infrastructure/database/_index.md:69]: missing quotes

File: docs-source/spring/content/infrastructure/database/_index.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+### ADB-S
```
Comment (markxnelson) [inline docs-source/spring/content/infrastructure/database/_index.md:71]: don't use abbreviation in heading

File: docs-source/spring/content/infrastructure/database/_index.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+   - `BYO ADB-S Compartment` : The compartment of the existing ADB-S.
```
Comment (markxnelson) [inline docs-source/spring/content/infrastructure/database/_index.md:73]: compartment name or ocid? if it is not obvious from the form

File: docs-source/spring/content/infrastructure/networking/_index.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
+---
+title: "Networking"
+description: "Networking Options and Configuration"
+keywords: "network spring springboot microservices oracle"
+---
+The Oracle Backend for Spring Boot and Microservices has specific networking requirements to ensure resource communication while providing security through isolation and networking rules.  
+
+The **standard installation** will provision a new Virtual Cloud Network (VCN) with the required subnets all networking rules to get started using the Oracle Backend for Spring Boot and Microservices quickly.  To use an existing VCN, please follow the [Bring Your Own VCN](#bring-your-own-oci-vcn) instructions.
```
Comment (markxnelson) [inline docs-source/spring/content/infrastructure/networking/_index.md:8]: missing "and" after "required subnets"?

File: docs-source/spring/content/infrastructure/networking/_index.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+> For **custom installations**, including On-Premises, it is the responsibility of the customer to ensure network access controls to provide both operational access and security.  The Oracle Cloud Infrastructure (OCI) **standard installation** setup can be used as a general template.
+
+# Network Infrastructure Overview
+
+![OCI Network Infrastructure](images/network_infra.png "OCI Network Infrastructure")
+
+## OCI Network Infrastructure
+
+All infrastructure resources are split between two subnets: one public and one private subnet with access restricted by Network Security Groups (Firewalls).  All access is blocked by default with specific IP and Port opened, as documented below, for proper operation.
```
Comment (markxnelson) [inline docs-source/spring/content/infrastructure/networking/_index.md:18]: port should not have capital letter

File: docs-source/spring/content/infrastructure/networking/_index.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+> Note minimum CIDRs are based on an Oracle Cloud Infrastructure (OCI) deployments.  Other cloud vendors reserve usable IPs within a CIDR reducing the usable IPs that can be used by the infrastructure.
+
+### Public Subnet
+
+The public subnet allows external access to the Oracle Backend for Spring Boot and Microservices resources.  Ingress/Egress to the Public Subnet and from the Internet is provided by an Internet Gateway.
+
+At a minimum, the public subnet requires two usable IP Addresses (`/30`).
```
Comment (markxnelson) [inline docs-source/spring/content/infrastructure/networking/_index.md:26]: expalin what the /30 means

File: docs-source/spring/content/infrastructure/networking/_index.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+ICMP traffic is allowed between the Public and Private Subnets for resource discovery. 
+
+#### Load Balancer
+
+An external Load Balancer is used in conjunction with Ingress resources to expose Services to the internet.  Ingress to the Load Balancer should be restricted by CIDR and Ports (default: `0.0.0.0/0`; `80`,`443`).  All TLS communications terminate at the LoadBalancer.  For more information on the certificate for TLS, please follow [OCI Load Balancer TLS Certificate Setup](#oci-loadbalancer-tls-certificate-setup).
```
Comment (markxnelson) [inline docs-source/spring/content/infrastructure/networking/_index.md:32]: replace "follow" with "see"

File: docs-source/spring/content/infrastructure/networking/_index.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+The Kubernetes API Endpoint, within the Control Plane, is used to manage the Kubernetes Cluster as well as providing access to Kubernetes services that are not exposed to the Load Balancer via Port Forwarding.  Ingress to the Endpoint is restricted to port `6443` and should be further restricted by CIDR (default: `0.0.0.0/0`).
+
+In Oracle Cloud Infrastructure (OCI), the Oracle Kubernetes Engine (OKE) Control Plane is allowed egress to OCI services via port `443`.  Access to these services are required for the OKE Managed Control Plane.
+
+### Private Subnet
+
+The private subnet isolates its resources from direct external access.  Ingress to resources in the private subnet is restricted to the Load Balancer and Kubernetes API Endpoint via specific Network Security Group (Firewall) rules.  Optionally, a Bastion Service, within the Private subnet, can provide additional non-restricted access to the resources in the Private Network.
+
+Egress to the Internet is provided by a NAT Gateway, while Egress to other Cloud Services (such as a Container Repository) is provided by a Service Gateway.
+
```
Comment (markxnelson) [inline docs-source/spring/content/infrastructure/networking/_index.md:45]: right about now i started wishing for a diagram of this :)

File: docs-source/spring/content/infrastructure/networking/_index.md [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+The CIDR of the Private Subnet is dependent on the number of Kubernetes Work Nodes and other resources, such as databases.  It is recommended to specify a CIDR providing at least six usable IP Addresses (`/29`).
+
+ICMP traffic is allowed between the Public and Private Subnets for resource discovery. 
+
+#### Kubernetes Worker Nodes
+
+The Worker Nodes are allowed to communicate with other Worker Nodes on all ports.
```
Comment (markxnelson) [inline docs-source/spring/content/infrastructure/networking/_index.md:52]: worker nodes - no caps

File: docs-source/spring/content/infrastructure/networking/_index.md [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+
+Worker Nodes are allowed to communicate with the API Endpoint on port `6443`.
+
+#### Worker Nodes and Control Plane
+
+The Kubernetes Control Plane is allowed to communicate to Worker Nodes on all ports.  The Worker Nodes are allowed to communicate with the Control Plane on port `12250`.
+
+#### Oracle Database
+
+It is recommended to place Oracle Databases inside the Private Subnet.  Worker Nodes and the Oracle Database will be allowed to communicate freely within the Private Subnet.
```
Comment (markxnelson) [inline docs-source/spring/content/infrastructure/networking/_index.md:68]: to place the Oracle Database....

File: docs-source/spring/content/infrastructure/networking/_index.md [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+
+## Configuration
+
+During the configuration of the Oracle Backend for Spring Boot and Microservices, ensure that the **Edition** is set to **Standard**:
+
+![Standard Edition](../images/standard_edition.png "Standard Edition")
+
+Tick the "Bring Your Own Virtual Network" checkbox and fill in the appropriate values for the VCN Compartment and Name, Public Subnet Compartment and Name, and Private Subnet and Name:
+
+   - `Virtual Cloud Network Compartment` : The compartment of the existing VCN.
```
Comment (markxnelson) [inline docs-source/spring/content/infrastructure/networking/_index.md:96]: name or ocid, if it is not obvious

File: docs-source/spring/content/setup/_index.md [hunk 12]
```diff
@@ -168,56 +175,62 @@ To set up the OCI environment, process these steps:
...
+      <!-- spellchecker-enable -->
+
+1. (*Standard Edition Only*) If you check the checkbox *Bring Your Own Virtual Cloud Network* in the **Network Options** section you can use an existing Virtual Cloud Network.  This is required to Bring Your Own Database (*Standard Edition Only*).
+
+      <!-- spellchecker-disable -->
+      {{< img name="oci-stack-network-options" size="large" lazy=false >}}
+      <!-- spellchecker-enable -->
 
-1. In the **Node Pool** section, fill in the following for the OKE Node Pools:
+    > For more information on the network requirements and topology of the Oracle Backend for Spring Boot and Microservices including the options for *Bring Your Own Virtual Cloud Network*, please review the [Networking](../infrastructure/networking) documentation.
```
Comment (markxnelson) [inline docs-source/spring/content/setup/_index.md:219]: please see the networking doc...

File: docs-source/spring/content/infrastructure/networking/_index.md (hunk 8 above)
Comment (gotsysdba) [reply docs-source/spring/content/infrastructure/networking/_index.md:47]: more diagram apart from the one at the top?

File: docs-source/spring/content/infrastructure/database/_index.md [hunk 13]
```diff
@@ -0,0 +1,101 @@
+---
+title: "Database"
+description: "Database Options and Configuration"
+keywords: "database spring springboot microservices oracle"
+
+---
+The Oracle Backend for Spring Boot and Microservices use the Oracle Database as a persistent data store for metadata and the Spring Cloud Config Server.  This documentation will refer to this database as the **Metadata Database**.
+
+> **NOTE:** Oracle recommends that you install an addition Container Database (CDB) and Pluggable Databases (PDBs) for your production applications inline with the Database-Per-Service pattern.  This document will refer to these databases as the [**Application Database**](#application-databases).
```
[the remark this answers is not in the document]
Comment (gotsysdba) [reply docs-source/spring/content/infrastructure/database/_index.md:9]: ok

File: docs-source/spring/content/infrastructure/database/_index.md [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+title: "Database"
+description: "Database Options and Configuration"
+keywords: "database spring springboot microservices oracle"
+
+---
+The Oracle Backend for Spring Boot and Microservices uses the Oracle Database as a persistent data store for metadata and the Spring Cloud Config Server.  This documentation will refer to this database as the **Metadata Database**.
+
+> **NOTE:** Oracle recommends that you install an addition Container Database (CDB) and Pluggable Databases (PDBs) for your production applications inline with the Database-Per-Service pattern.  This document will refer to these databases as the [**Application Database**](#application-databases).  
+
+By default, the Oracle Autonomous Database - Shared (ADB-S) is used for the  **Metadata Database**, however, there are other options including Bring Your Own (BYO).
```
Comment (markxnelson) [inline docs-source/spring/content/infrastructure/database/_index.md:11]: The "S" is "Serverless" now, not "Shared"

File: docs-source/spring/content/infrastructure/database/_index.md [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+GRANT CREATE ANY TABLE TO OBAAS;
+GRANT INSERT ANY TABLE TO OBAAS;
+GRANT SELECT ANY TABLE TO OBAAS;
+GRANT UPDATE ANY TABLE TO OBAAS;
+GRANT CREATE ANY SEQUENCE TO OBAAS;
+GRANT SELECT ANY SEQUENCE TO OBAAS;
+GRANT CREATE ANY TRIGGER TO OBAAS;
+-- Additional for PARSE
+GRANT SODA_APP TO OBAAS WITH ADMIN OPTION;
+GRANT CREATE TABLE TO OBAAS WITH ADMIN OPTION;
```
Comment (markxnelson) [inline docs-source/spring/content/infrastructure/database/_index.md:60]: does this user need aq_adm.... as well?

**Review by markxnelson — APPROVED** [review]
please see a couple of very minor comments

File: docs-source/spring/content/infrastructure/database/_index.md (hunk 15 above)
Comment (gotsysdba) [reply docs-source/spring/content/infrastructure/database/_index.md:60]: @andytael .. do you have the AQ stuffs?

Comment (andytael) [reply docs-source/spring/content/infrastructure/database/_index.md:60]: @gotsysdba here it is:
```
grant aq_user_role to account;
grant execute on dbms_aq to account;
grant execute on dbms_aqadm to account;
grant execute on dbms_aqin to account;
grant execute on dbms_aqjms_internal to account;
```
Don't know why `dbms_aqjms_internal` is there?

**Review by andytael — APPROVED** [review]
LGTM

**Pull request merged**